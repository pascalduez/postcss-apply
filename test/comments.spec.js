import postcss from 'postcss';
import plugin from '../src';

const input = `
:root {
  /* This should be pruned */
  --toolbar-theme: {
    color: orangeRed;
  }
}

.toolbar {
  @apply --toolbar-theme;
}
`;

let from;

describe('comments', () => {
  it('should remove immediate preceding comments in declarations', async () => {
    const result = await postcss()
      .use(plugin)
      .process(input, { from });

    expect(result.css).toMatchSnapshot();
  });

  it('should not remove immediate preceding comments in declarations with the preserve option', async () => {
    const result = await postcss()
      .use(plugin({ preserve: true }))
      .process(input, { from });

    expect(result.css).toMatchSnapshot();
  });
});
