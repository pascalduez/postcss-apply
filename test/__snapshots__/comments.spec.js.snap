// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comments should not remove immediatly preceding comments in declarations with the preserve option 1`] = `
"
:root {
  /* This should be pruned */
  --toolbar-theme: {
    color: orangeRed;
  }
}

.toolbar {
  color: orangeRed;
  @apply --toolbar-theme;
}
"
`;

exports[`comments should properly remove immediatly preceding comments in declarations 1`] = `
"
.toolbar {
  color: orangeRed;
}
"
`;
